Possible Bug due to recent PR

Hello there, I started seeing unexpected issues in one of my CircleCI builds that uses gotestsum, and I'm not sure what's causing it (previously the same commits were passing).

One possible theory is that there is an error with a recent PR: https://github.com/gotestyourself/gotestsum/pull/79. Here's why I suspect that:
* The error I get is from `testjson/dotformat.go:73` (full stack trace and screenshot below). 
* If I run the CircleCI build with ssh and run the test command there, everything passes. [This code](https://github.com/gotestyourself/gotestsum/blob/master/testjson/dotformat.go#L71) seems to have something to do with the size of the terminal, so perhaps it is a difference about whether or not the tests are being run from a shell?
* Specifying a specific version `v0.4.0` of gotestsum (which I should have been doing anyway) seems to have fixed it.

Feel free to close this issue if it does not seem relevant/ is not helpful.

Stack trace:
```sh
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x5898e7]

goroutine 1 [running]:
gotest.tools/gotestsum/testjson.newDotFormatter(0x6873c0, 0xc000010018, 0x1, 0xc000000180)
	/home/circleci/.go_workspace/src/gotest.tools/gotestsum/testjson/dotformat.go:73 +0x67
gotest.tools/gotestsum/testjson.NewEventFormatter(0x6873c0, 0xc000010018, 0x7fff7f97f5c2, 0x4, 0x5ea440, 0xc00000e2c0)
	/home/circleci/.go_workspace/src/gotest.tools/gotestsum/testjson/format.go:202 +0x1a4
main.newEventHandler(0xc000028100, 0x6873c0, 0xc000010018, 0x6873c0, 0xc000010020, 0xc0000ca160, 0x6873a0, 0xc000010060)
	/home/circleci/.go_workspace/src/gotest.tools/gotestsum/handler.go:51 +0x53
main.run(0xc000028100, 0x0, 0x0)
	/home/circleci/.go_workspace/src/gotest.tools/gotestsum/main.go:139 +0x133
main.main()
	/home/circleci/.go_workspace/src/gotest.tools/gotestsum/main.go:39 +0x154
```

<img width="1008" alt="Continuous_Integration_and_Deployment" src="https://user-images.githubusercontent.com/57721310/71397590-fe2b1500-25eb-11ea-99e2-1e2253fea4a0.png">

--- a/testjson/dotformat.go
+++ b/testjson/dotformat.go
@@ -70,8 +70,7 @@ func (l *dotLine) checkWidth(prefix, terminal int) {
 func newDotFormatter(out io.Writer) EventFormatter {
 	w, _, err := terminal.GetSize(int(os.Stdout.Fd()))
 	if err != nil || w == 0 {
-		logrus.Warn(err.Error())
-		logrus.Warn("Failed to detect terminal width for dots format.")
+		logrus.Warnf("Failed to detect terminal width for dots format, error: %v", err)
 		return &formatAdapter{format: dotsFormatV1, out: out}
 	}
 	return &dotFormatter{

--- a/testjson/dotformat_test.go
+++ b/testjson/dotformat_test.go
@@ -12,11 +12,11 @@ import (
 	"time"
 	"unicode/utf8"
 
-	"gotest.tools/gotestsum/internal/dotwriter"
-
 	"gotest.tools/assert"
 	"gotest.tools/assert/cmp"
 	"gotest.tools/golden"
+	"gotest.tools/gotestsum/internal/dotwriter"
+	"gotest.tools/skip"
 )
 
 func TestScanTestOutput_WithDotsFormatter(t *testing.T) {
@@ -151,4 +151,13 @@ func TestFmtDotElapsed_RuneCountProperty(t *testing.T) {
 		MaxCountScale: 2000,
 		Rand:          rand.New(rand.NewSource(seed)),
 	}))
+}
+
+func TestNewDotFormatter(t *testing.T) {
+	buf := new(bytes.Buffer)
+	ef := newDotFormatter(buf)
+
+	d, ok := ef.(*dotFormatter)
+	skip.If(t, !ok, "no terminal width")
+	assert.Assert(t, d.termWidth != 0)
 }

